General maintenance and updated dependencies, to help bring this package up-to-date

JackWH opened the pull request:
After being held up by issue #40, I've gone ahead and updated the dependencies used by this package to bring it up to date.

Here's what's changed:

- Updated `composer.json` to use version `^2.0` of `themsaid/forge-sdk`. Version 2.0 has some new parameters added to certain functions ([see here](https://github.com/themsaid/forge-sdk/releases/tag/v2.0)), which I haven't accounted for in `forge-cli` — but the function definitions remain backwards-compatible.
- Updated `composer.json` to use version `^5.0` of `symfony/console` and `symfony/process`. This was necessary as the current release of `laravel/homestead` installs version `^5.0` of these packages globally, so the existing dependency version `^4.0` was causing conflicts that prevented installation.
- Added `return 0;` to every `symfony/console` inherited `execute()` function, a new requirement following the version updates. `return 0;` indicates the command ran successfully, any other result should be an integer and will be returned as a process exit status to the shell.
- Changed the `forge authorize` command to allow for the API key to be passed in as an optional parameter. If no parameter is specified, the user will be prompted to type it in to the terminal window themselves. This change allows programmatic setup of the Forge CLI via other Artisan commands.

svenluijten commented on composer.json:
```diff
@@ -12,9 +12,9 @@
-        "symfony/process": "^4.0",
-        "themsaid/forge-sdk": "^1.2",
+        "symfony/console": "^5.0",
+        "symfony/process": "^5.0",
```
```suggestion
        "symfony/process": "^4.0 || ^5.0",
```

I'd rather keep these like so to still _allow_ for `4.x` to be installed. Otherwise we'll get conflicts with people who don't have the latest version of Homestead 🙂

svenluijten commented on composer.json:
```diff
@@ -12,9 +12,9 @@
-        "symfony/console": "^4.0",
-        "symfony/process": "^4.0",
-        "themsaid/forge-sdk": "^1.2",
+        "symfony/console": "^5.0",
```
```suggestion
        "symfony/console": "^4.0 || ^5.0",
```

jackwh commented:
@svenluijten Thanks, happy to contribute! I just pushed a few more tweaks to this branch...

- Changed dependency version constraints, as you suggested
- This one is kind of opinionated, but I noticed the package just dumps a `forge.json` config file in the user's home directory:
  * I changed the `getFileConfig()` function to prefer a hidden `~/.forge.json` file instead
  * It will continue to use `~/forge.json` if one exists for backwards-compatibility
  * Going forward, new calls to `forge authorize` will save a hidden file instead
  * I feel this is more in keeping with convention for global packages
- When a `.forge.json` configuration file is created for the first time, a JSON `key` property is now being set to an empty string:
  - Previously, it was just an empty object being saved (`{}`)
  - This change ensures "Unauthorized" exceptions are thrown by Guzzle, instead of a PHP fatal error
  - Errors would be thrown when trying to call the Forge API before `forge authorize` has been run

svenluijten commented:
So sorry for the delay on this, but due to what's going on in the world haven't really had the time to check this out in-depth. I will get to it this weekend, I promise!

barryvdh commented:
This weekend? ;)
I'm looking into managing backups. That requires https://github.com/themsaid/forge-sdk/pull/67 to be merged but then it would need v2 of the SDK.

svenluijten commented:
Tagged as [`v0.5.0`](https://github.com/svenluijten/forge-cli/releases/tag/v0.5.0)!

svenluijten commented:
Btw @barryvdh, I'd be happy to get those backup methods also integrated into this CLI once it makes its way into the SDK. 👍
